src/components/UserProfileDetails/UserProfileDetails.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+  ChangeEvent,
+  FormEvent,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import Card from '@mui/material/Card';
+import CardActions from '@mui/material/CardActions';
+import CardContent from '@mui/material/CardContent';
+import CardHeader from '@mui/material/CardHeader';
+import Divider from '@mui/material/Divider';
+import TextField from '@mui/material/TextField';
+import Grid from '@mui/material/Unstable_Grid2';
+import { ProfileSliceState } from '../../redux/profile/slice';
+import { useAppDispatch } from '../../app/hooks';
+import { updateUserProfile } from '../../redux/profile/operations';
+import { TextareaAutosizeStyled } from './UserProfileDetails.styled';
+import { toast } from 'react-toastify';
+
+export const UserProfileDetails = ({
+  profile,
+}: {
+  profile: ProfileSliceState;
+}) => {
+  const dispatch = useAppDispatch();
+  const [values, setValues] = useState<ProfileSliceState>({
```

Comment:
for cases where you have not loaded or nonexisting valeus, it is better to use null
like `const [values, setValues] = useState<ProfileSliceState | null>({`

Reply:
have a problem with that